src/server/pachyderm_test.go

```diff
@@ -363,6 +363,107 @@ func TestPipeline(t *testing.T) {
 	require.Equal(t, 2, len(listCommitResponse.CommitInfo))
 }
 
+func TestPipelineWithTooMuchParallelism(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	t.Parallel()
+	pachClient := getPachClient(t)
+	// create repos
+	dataRepo := uniqueString("TestPipelineWithTooMuchParallelism.data")
+	require.NoError(t, pfsclient.CreateRepo(pachClient, dataRepo))
+	// create pipeline
+	pipelineName := uniqueString("pipeline")
+	outRepo := ppsserver.PipelineRepo(ppsclient.NewPipeline(pipelineName))
+	// This pipeline will fail if any pod sees empty input, since cp won't
+	// be able to find the file.
+	// We have parallelism set to 3 so that if we actually start 3 pods,
+	// which would be a buggy behavior, some jobs don't see any files
+	require.NoError(t, ppsclient.CreatePipeline(
+		pachClient,
+		pipelineName,
+		"",
+		[]string{"cp", path.Join("/pfs", dataRepo, "file"), "/pfs/out/file"},
+		nil,
+		3,
+		[]*ppsclient.PipelineInput{{
+			Repo:   &pfsclient.Repo{Name: dataRepo},
+			Reduce: true, // setting reduce to true so only one pod gets the file
+		}},
+	))
+	// Do first commit to repo
+	commit1, err := pfsclient.StartCommit(pachClient, dataRepo, "", "")
+	require.NoError(t, err)
+	_, err = pfsclient.PutFile(pachClient, dataRepo, commit1.ID, "file", strings.NewReader("foo\n"))
+	require.NoError(t, err)
+	require.NoError(t, pfsclient.FinishCommit(pachClient, dataRepo, commit1.ID))
+	listCommitRequest := &pfsclient.ListCommitRequest{
+		Repo:       []*pfsclient.Repo{outRepo},
+		CommitType: pfsclient.CommitType_COMMIT_TYPE_READ,
+		Block:      true,
+	}
+	listCommitResponse, err := pachClient.ListCommit(
+		context.Background(),
+		listCommitRequest,
+	)
```

Comment:
Would you mind doing an `InspectJob` here like in the other tests? Also add a timeout to the context so that the test can error after like 30 seconds rather than blocking indefinitely and causing a timeout.